lib/ruby_lsp/requests/document_link.rb

```diff
@@ -50,8 +50,7 @@ def gem_paths
               spec = stub.to_spec
               lookup[spec.name] = {}
               lookup[spec.name][spec.version.to_s] = {}
-              prefix_matchers = [//]
-              prefix_matchers.concat(spec.require_paths.map { |rp| Regexp.new("^#{rp}/") })
+              prefix_matchers = spec.require_paths.map { |rp| Regexp.new("^#{rp}/") }
```

Comment:
In Ruby 3.2, the `mutex_m.rb` spec file entry moved to `lib/mutex_m.rb`, but our regex was actually not properly removing the `lib/` part.